frontend/catalyst/jax_tracer.py

```diff
@@ -528,11 +528,27 @@ def trace_quantum_function(
         # (1) - Classical tracing
         quantum_tape = QuantumTape()
         with EvaluationContext.frame_tracing_context(ctx) as trace:
-            wffa, in_avals, out_tree_promise = deduce_avals(f, args, kwargs)
+            wffa, in_avals, _ = deduce_avals(f, args, kwargs)
             in_classical_tracers = _input_type_to_tracers(trace.new_arg, in_avals)
             with QueuingManager.stop_recording(), quantum_tape:
                 # Quantum tape transformations happen at the end of tracing
                 ans = wffa.call_wrapped(*in_classical_tracers)
+
+            # Ans contains the leaves of the pytree (empty for measurement without
+            # data https://github.com/PennyLaneAI/pennylane/pull/4607)
+            # Therefore we need to compute the tree with measurements as leaves and it comes
+            # with an extra computational cost
+
+            # 1. Recompute the original return
+            with QueuingManager.stop_recording():
+                ans = tree_unflatten(wffa.stores[0].val, ans)
```

Comment:
If the tree in `wffa.stores` is the same as the tree in `out_tree_promise`, maybe we can just it here?
```suggestion
                ans = tree_unflatten(out_tree_promise, ans)
```

Reply:
Updated :+1: